chain/epoch_manager/src/validator_selection.rs

```diff
@@ -684,14 +711,69 @@ mod tests {
         // too low stakes are kicked out
         let kickout = epoch_info.validator_kickout();
         assert_eq!(kickout.len(), 2);
+        #[cfg(feature = "protocol_feature_fix_staking_threshold")]
+        let expected_threshold = 334;
+        #[cfg(not(feature = "protocol_feature_fix_staking_threshold"))]
+        let expected_threshold = 300;
         assert_eq!(
             kickout.get("test5").unwrap(),
-            &ValidatorKickoutReason::NotEnoughStake { stake: 100, threshold: 300 },
+            &ValidatorKickoutReason::NotEnoughStake { stake: 100, threshold: expected_threshold },
         );
         assert_eq!(
             kickout.get("test6").unwrap(),
-            &ValidatorKickoutReason::NotEnoughStake { stake: 50, threshold: 300 },
+            &ValidatorKickoutReason::NotEnoughStake { stake: 50, threshold: expected_threshold },
         );
+
+        #[cfg(feature = "protocol_feature_fix_staking_threshold")]
+        {
+            let bp_threshold = epoch_info.seat_price();
+            let num_validators = epoch_info.validators_iter().len();
+            let proposals = create_proposals(&[
+                ("test1", 1000),
+                ("test2", 1000),
+                ("test3", 1000), // the total up to this point is 3000
+                ("test4", 200),  // 200 is < 1/10 of 3000, so not validator, but can be fisherman
+                ("test5", 100),  // 100 is even too small to be a fisherman, cannot get any role
+                ("test6", 50),
+                ("test7", bp_threshold),
+            ]);
+            let epoch_info = proposals_to_epoch_info(
+                &epoch_config,
+                [0; 32],
+                &epoch_info,
+                proposals,
+                Default::default(),
+                Default::default(),
+                0,
+                PROTOCOL_VERSION,
+                PROTOCOL_VERSION,
+            )
+            .unwrap();
+            assert_eq!(num_validators + 1, epoch_info.validators_iter().len());
```

Comment:
Not sure if it is worth it for this case, but, in general, I'd prefer to `cfg` *just* the asserts, rather than the whole test, such that the single test checks both the old and the new behavior.